src/components/font-family-control/index.js

```diff
@@ -40,11 +61,11 @@ const FontFamilyControl = props => {
 				options: fontOptions,
 			},
 		] )
-	}, [] )
+	}, [ themeFonts ] )
 
 	return (
 		<AdvancedAutosuggestControl
-			options={ options }
+			options={ themeFonts.length ? options : options.filter( option => option.id !== 'theme-fonts' ) }
```

Comment:
Do not filter it out here, instead, the theme fonts should not be added from the start if there aren't any